iwyu_include_picker.cc

```diff
@@ -1337,13 +1340,22 @@ vector<IncludePicker::MappedInclude>
 IncludePicker::GetCandidateHeadersForFilepath(
     const string& filepath, const string& including_filepath) const {
   CHECK_(has_called_finalize_added_include_lines_ && "Must finalize includes");
-  const string quoted_header = ConvertToQuotedInclude(
-      filepath, MakeAbsolutePath(GetParentPath(including_filepath)));
+  string quoted_header;
+  if (including_filepath.empty()) {
+    quoted_header = ConvertToQuotedInclude(filepath);
+  } else {
+    quoted_header = ConvertToQuotedInclude(
+        filepath, MakeAbsolutePath(GetParentPath(including_filepath)));
+  }
   vector<MappedInclude> retval =
       GetPublicValues(filepath_include_map_, quoted_header);
-  if (retval.empty()) {
-    // the filepath isn't in include_map, so just quote and return it.
-    retval.push_back(MappedInclude(quoted_header, filepath));
+
+  // We also need to consider the header itself.  Make that an option if it's
+  // public or there's no other option.
+  MappedInclude default_header(quoted_header, filepath);
+  if (retval.empty() || GetVisibility(default_header) != kPrivate) {
```

Comment:
Ah, is this where the map to self thing went? It should default to `kPublic` here too. Oh, wait. `kPublic` is not the same as `kPrivate`, and neither is `kUnusedVisibility`, so I guess this will work as long as the predicates are phrased exactly like this. I think I would feel more comfortable with `kPublic` being spelled out here.

Reply:
Done.